### PR TITLE
WIP | Initial help with documentation

### PR DESCRIPTION
**Pull request by courtyenn**
Organizing and adding more content to Kremling's docs.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
-and requires no changes to your webpack config or build process.
+Kremling is an npm library for writing CSS with React. It is only 4kb gzipped, and requires no changes to your webpack config or build process.
 
 [Kremling Github page](https://github.com/CanopyTax/kremling)
 
 [Walkthrough](/walkthrough/getting-started.md)
 
-## Hooks example
+## Motivation
+Why use Kremling over say, CSS Modules?
```
Comment (internettrans) [inline docs/README.md:9]: I would rather not call out specific competitor libraries and try to justify why kremling is better than them (since I honestly don't really believe that kremling is actually better than every other css lib option). Instead, I'd rather just focus on why we built kremling.

So I'd prefer to remove this sentence that calls out CSS modules

File: docs/README.md [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+
+1. **Option to use [React hooks](https://reactjs.org/hooks)**. Kremling is bleeding edge. :)
```
Comment (internettrans) [inline docs/README.md:11]: I'd rather not have hooks listed as a motivation for kremling, but a feature of kremling.

Bleeding edge is not really true since hooks are now part of react's official latest release. Additionally, I don't think that riding the bleeding edge is actually a good reason to use something. And hooks won't be new for very long, so there's a chance that this documentation becomes stale and feels out of date within a few months.

File: docs/README.md [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+2. **Readable class names**. In your browser dev tools, `<div class="card">` is easier to understand than `<div class="23fgh_es56_card">`.
+3. **Make it easy to [conditionally apply css](walkthrough/conditional-css.md)**. We can do better than ternaries for changing a DOM element's CSS classes.
+4. **Remove unused CSS from the DOM**. When there are no more components that are using a CSS class, the `<style>` element for those components should be removed from the DOM.
+5. [**Scoped css**](concepts/scoped-css.md). Have CSS rules apply to a component and its children, but no other components. Allow for cascading rules within a "scope".
+
+
+## Hooks
+Hooks are a part of React and Kremling likes them because it makes code extremely functional. (Joel help)
```
Comment (internettrans) [inline docs/README.md:19]: Why do we need this sentence or the Hooks section?

File: docs/README.md [hunk 4]
```diff
@@ -35,7 +47,7 @@ const css = `
 `
 ```
 
-## Class component example
+### Example without hooks
```
Comment (internettrans) [inline docs/README.md:50]: Why did you change this?

I originally had this header say "Example without hooks," but I changed it to "Class component example" because I think that for a user of kremling the question is whether they can use this with their class components.

I don't think the header will make a big difference either way -- I'm just curious why you changed it.

File: docs/README.md (hunk 3 above)
Comment (courtyenn) [reply docs/README.md:19]: I think it is very unique to Kremling and CSS libraries. From a use POV, I'd like to understand  how it solves a problem. If there is? Or if Kremling thinks it is super cool I think it would be nice to explain a TLDR; about them.

File: docs/README.md (hunk 4 above)
Comment (courtyenn) [reply docs/README.md:50]: I think hooks are an intimidating topic while classes are more foundational. I named it this so users don't feel obligated to use hooks with Kremling.

File: docs/README.md (hunk 3 above)
Comment (internettrans) [reply docs/README.md:19]: It won't be unique about kremling for long 😄. Also, the section right below this section is called Example with Hooks. Is that enough for people to see that we're using hooks?

File: docs/README.md (hunk 2 above)
Comment (courtyenn) [reply docs/README.md:11]: Ok. I agree with this. I wanted to follow up with some type of comment for why Kremling worked to support hooks.
